.gitlab-ci.yml

```diff
@@ -400,10 +400,9 @@ pkg:opam:
     - cat /proc/{cpu,mem}info || true
     # Use current worktree as tmpdir to allow exporting artifacts in case of failure
     - export TMPDIR=$PWD
-    # We build an expression rather than a direct URL to not be dependent on
-    # the URL location; we are forced to put the public key of cache.nixos.org
+    # We are forced to put the public key of cache.nixos.org
     # because there is no --extra-trusted-public-key option.
-    - nix-build -E "import (fetchTarball $CI_PROJECT_URL/-/archive/$CI_COMMIT_SHA.tar.gz) {}" -K --extra-substituters "$EXTRA_SUBSTITUTERS" --trusted-public-keys "$NIXOS_PUBLIC_KEY $EXTRA_PUBLIC_KEYS" | if [ ! -z "$CACHIX_SIGNING_KEY" ]; then cachix push coq; fi
+    - nix-build -K --extra-substituters "$EXTRA_SUBSTITUTERS" --trusted-public-keys "$NIXOS_PUBLIC_KEY $EXTRA_PUBLIC_KEYS" | if [ ! -z "$CACHIX_SIGNING_KEY" ]; then cachix push coq; fi
```

Comment:
This builds `coq-full`, so this tests just as much as before. But we could also build `coq` so that it is available through Cachix and developers do not have to rebuild it locally when preparing overlays.
(Later, we could even go further and change the whole CI infrastructure to rely on the Coq Nix toolbox.)